account-model/src/account/account_builder.rs

```diff
@@ -0,0 +1,317 @@
+            organisation_id: get_or_create_id(self.organisation_id),
+            country: CountryCode::GBR,
+            currency: Currency::GBP,
+            bank_id: self.bank_id.as_ref().expect("bank id is required").to_string(),
+            bank_id_code: self.bank_id_code.as_ref().expect("bank id code is required").to_string(),
+            bic: self.bic.as_ref().unwrap().to_string(),
+            number: self.number.clone(),
+            iban: self.iban.clone(),
+            title: self.title.clone(),
+            classification: self.classification,
+        })
+    }
+
+    fn validate_account_restriction_by(&mut self,
+                                       country: CountryCode,
+                                       expected_number_length: u8,
+                                       expected_bank_id_length: u8,
+                                       expected_code: KnownBankIdCode) -> Result<(), String> {
+        if self.number.is_some() {
+            if self.number.as_ref().unwrap().len() != expected_number_length as usize {
+                return Err(format!("{} requires {}-character long Account Number",
+                                   country.alpha3(), expected_number_length));
+            }
+        }
+        if self.bank_id.is_none() || self.bank_id.as_ref().unwrap().len() != expected_bank_id_length as usize {
+            return Err(format!("{} requires {}-character BankId, got '{:?}'",
+                               country.alpha3(), expected_bank_id_length, self.bank_id));
+        }
+
+        let bank_id_code: &str = expected_code.into();
+        let bank_id_code_s: &String = &bank_id_code.to_string();
+        if self.bank_id_code.is_none() || self.bank_id_code.as_ref().unwrap().ne(bank_id_code_s) {
+            debug!(target: "account_builder", "{} requires BankIdCode, got invalid '{:?}'",
+                   country.alpha3(), self.bank_id_code);
+            self.bank_id_code = Some(bank_id_code_s.to_string());
+        }
+        return Ok(());
+    }
+}
+
+fn get_or_create_id(given: Option<Uuid>) -> Uuid {
+    given.unwrap_or_else(|| Uuid::new_v4())
+}
+
+#[cfg(test)]
+mod tests {
+    use iso_currency::Currency::GBP;
+    use isocountry::CountryCode::GBR;
+    use uuid::Uuid;
+
+    use crate::account::Classification::{BUSINESS, PERSONAL};
+    use crate::account::KnownBankIdCode;
+
+    use super::Builder;
+
+    #[test]
+    fn should_fail_as_no_country() {
+        let actual = Builder::new_account()
+            .build()
+            .unwrap_err();
+
+        // then
+        assert_eq!("Country is required", actual);
+    }
+
+    macro_rules! invalid_account_parametrised {
+        ($($name:ident: $value:expr,)*) => {
+            $(
+                #[test]
+                fn $name() {
+                    let (country, bic, number, bank_id, expected) = $value;
+                    let mut builder: Builder = Builder::new_account();
+                    let builder: &mut Builder = &mut builder;
```

Comment:
tutaj też zrobiłem jakąś sieczkę :D

Reply:
jeśli tego nie widzisz to chodzi mi o to ze metody zwrcają `&mut buidler` ale nie moge zrobić po porstu 
```
var builder = Builder::new_account()
                        .with_country(country)
                        .mark_business();
if (cos tam) {
   builder.....
}
```

Reply:
albo ja czegoś nie widzę